Adding feature to make last used config be saved and used as the default

ahobbs7 opened the pull request:
When a config is loaded and used, that config is now saved to the napari configuration directory, meaning that it will be used in future instances of napari when a config is not loaded, rather than using the generated default config. Additionally, the file explorer is now opened the moment load config is pressed to save an extra click!

llwiggins commented:
Hey @ahobbs7 this code runs error-free! I just wanted to check I've understood what the aim is here. 

So here "Run Filters" and "Run Grains" make use of the config file stored in memory (which is the last config file imported into Napari) rather than using a default config? I'm trying to think of the drawbacks of this approach as users may be performing a completely new analysis, or may not have used napari for a long time and therefore their stored config could be quite outdated...?

Are there also pitfalls if we update TopoStats but the stored config is from an older version?

I'd be tempted to keep the default config as the one loaded in napari by default, but I'd be interested to hear experimentalists opinions on this!

ahobbs7 commented:
I just created a new PR #60 which should address some of the problems of saving a new default everytime a config is loaded by making that manual, and including the option to reset the default to the topostats generated one. I will add unaddressed issues raised by @ns-rse as issues :)
